src/main/java/seedu/cakecollate/model/order/ContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,119 @@
+        }
+        return result;
+    }
+
+    /**
+     * Test each {@code String} retrieved from {@code Order} to see if {@code String}
+     * contains any of the provided keywords.
+     */
+    public boolean prefixTest(List<String> toTest, String toTestFrom) {
+        return toTest.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(toTestFrom, keyword));
+    }
+
+    /**
+     * For the specified {@code Order}, retrieve, process and return the {@code String} to test keywords against
+     * based on the provided {@code Prefix}.
+     */
+    public String getTestString(Prefix prefix, Order order) {
+        // Cannot use switch-case unless I edit the Prefix class...
+        if (prefix.equals(PREFIX_NAME)) {
+            return order.getName().fullName;
+        } else if (prefix.equals(PREFIX_PHONE)) {
+            return order.getPhone().value;
+        } else if (prefix.equals(PREFIX_EMAIL)) {
+            return order.getEmail().value;
+        } else if (prefix.equals(PREFIX_ADDRESS)) {
+            return order.getAddress().value;
+        } else if (prefix.equals(PREFIX_ORDER_DESCRIPTION)) {
+            Set<OrderDescription> orderDescriptions = order.getOrderDescriptions();
+            Set<String> orderDescriptionsString = orderDescriptions.stream()
+                    .map(OrderDescription::toString)
+                    .collect(Collectors.toSet());
+            return String.join(" ", orderDescriptionsString);
+        } else if (prefix.equals(PREFIX_TAG)) {
+            Set<Tag> tags = order.getTags();
+            Set<String> tagsString = tags.stream()
+                    .map(tag -> tag.tagName)
+                    .collect(Collectors.toSet());
+            return String.join(" ", tagsString);
+        } else if (prefix.equals(PREFIX_DATE)) {
+            LocalDate deliveryDate = order.getDeliveryDate().value;
+            DateTimeFormatter format1 = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+            DateTimeFormatter format2 = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+            DateTimeFormatter format3 = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+            DateTimeFormatter format4 = DateTimeFormatter.ofPattern("dd MMM yyyy");
+            return deliveryDate.getDayOfMonth() + " "
+                    + deliveryDate.getMonthValue() + " "
+                    + deliveryDate.getMonth().toString() + " "
+                    + deliveryDate.getYear() + " "
+                    + format1.format(deliveryDate) + " "
+                    + format2.format(deliveryDate) + " "
+                    + format3.format(deliveryDate) + " "
+                    + format4.format(deliveryDate) + " ";
+        } else if (prefix.equals(PREFIX_REQUEST)) {
+            return order.getRequest().value;
+        } else if (prefix.equals(PREFIX_DELIVERY_STATUS)) {
+            return order.getDeliveryStatus().toString();
+        } else if (prefix.equals(new Prefix("all/"))) {
```

Comment:
Would using a PREFIX_ALL variable be less bug-prone in case you need to add future tests? Not very sure myself, I think as long as we assert the user doesn't use that prefix, it should be okay?

Reply:
Yeap my initial concern was I didn't want to let user be able to use this prefix at all thats why I only create it in the code. Just remember I can create the prefix but not set it as a prefix argmultimap accept haha thanks